git-config/src/file/git_config.rs

```diff
@@ -289,10 +294,25 @@ impl<'event> GitConfig<'event> {
             }
 
             let mut paths_to_include = Vec::new();
-            for path in target_config
+
+            let mut include_paths = target_config
                 .multi_value::<values::Path>("include", None, "path")
-                .unwrap_or_default()
-            {
+                .unwrap_or_default();
+
+            for (header, body) in target_config.sections_by_name_with_header("includeIf") {
+                if let Some(condition) = &header.subsection_name {
+                    if include_condition_is_true(condition, options) {
+                        let paths: Vec<values::Path> = body
+                            .values(&Key::from("path"))
+                            .iter()
+                            .map(|path| values::Path::from(path.clone()))
+                            .collect();
```

Comment:
No need to collect here - `extend()` below takes an iterator.

Reply:
ah, ok!